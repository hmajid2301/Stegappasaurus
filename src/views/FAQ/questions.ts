import { IFAQ } from "~/components/FAQList";

const faq: IFAQ[] = [
  {
    content:
      "Steganography is the art/science of hiding data in plain sight. In the context of this app, it allows " +
      "you to hide messages (text) within images. You do this by encoding the image with the text data. You can then share the image " +
      "with other people and they can decode the image (retrieve the original message) using this app.",
    title: "What is steganography?"
  },
  {
    content:
      "Not quite, encryption involves obscuring information whereas steganography is concerned with hiding it. " +
      "With encryption if you have the encrypted message you cannot retrieve the original message. \n\n" +
      "Whereas with steganography if you know where to look you can retrieve the original message, it is not obscured in any way. " +
      "However, with encryption it is obvious you are trying to hide something but with steganography you can hide your message inside " +
      'of an "innocent" looking image, with a third party without anyone noticing.\n\n' +
      'You can combine the two "techniques", encrypt your data before encoding it into an image, then simply decrypt the' +
      "message after decoding it from the image, " +
      'for added security. This involves a shared secret "password" between you and the person you are sending the image to.',
    title: "So is it the same as encryption?"
  },
  {
    content:
      "This app uses different steganography algorithms to encode the RGB (Red, Green, Blue) pixel values with the text " +
      "data in such a way that the original data can be retrieved at a later date.\n\n" +
      "After the image has been encoded you can save the image, then you can share this image and you use this app " +
      "to decode and get the originally encoded message.\n\n" +
      "In general each algorithm will encode the size of the message, this is so that when decoding the image we know when to stop. " +
      "It will then encode a separator (Â£ in ASCII), this is so that again when we decode we know from this point to decode n characters. " +
      "Each character in the message usually uses 8 bits in the images, this is because we encode it as a the ASCII representation of the letter or symbol.",
    title: "How does this app work?"
  },
  {
    content:
      "You can use the following algorithms to encode/decode images: \n\n" +
      "Bitmap Least Significant Bit (LSB) \n" +
      "Discrete Cosine Transform (DCT) Least Significant Bit (LSB) \n",
    title: "What Algorithms can you use?"
  },
  {
    content:
      "It all depends on the image resolution, the bigger the image the more data you can encode. " +
      "To find out if your message will fit in the image you should solve Equation 1 shown below where," +
      "\n\n h = Image Height (in Pixels), \n w = Image Width (in pixels) and \n m is the number of characters in your message\n\n" +
      'To find the maximum number of characters we can encode we must use the same equation above except we = 0 not > 0 and we will solve for "m". ' +
      "Let's say our image is 100 x 100, then we have 100 * 100 * 3 = 30,000 - 8 (for separator) = 29,992. So 29,992 = ceil(log2(m)) - 8m = 3747. " +
      "So using the LSB algorithm we can encode a message with 3747 characters.\n\n" +
      "Note: With LSB we only can encode one bit per byte of data, so that we don't completely change the colour of that pixel.\n\n" +
      "Equation 1:\n\n 3(h * w) - ceil(log2(m)) - 8m -8 > 0",
    title: "How much data can I encode with LSB?"
  },
  {
    content:
      "We will use an image 100 x 100 = 10,000 pixels. So 10,000 * 3 * 8 = 30,000 bits we can encode." +
      "Let's say we want to encode \"Hello World!\", it's of length 12 so ceil(log2(12)) = 4 bits for the length of the message. " +
      "So  4 + 8 = 12 bits required so far. Each character we encode will take 8 bits and we have 12 characters " +
      "(including blank space) 12 * 8 = 96 bits for the message, so in total 96 + 12 = 108 bits " +
      'would be required out of 30,000, to encode the message "Hello World!".',
    title: "Can you show me an example?"
  },
  {
    content:
      "If we used the entire byte, remember per pixel we get 3 bytes (one for each RGB item), we would drastically " +
      "change the colour of image.",
    title: "Why can we only use one bit out of every byte?"
  }
];

export default faq;
