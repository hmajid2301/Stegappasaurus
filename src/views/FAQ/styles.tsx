import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
  },

  faqListContainer: {
    alignItems: 'stretch',
    padding: 5,
  },
});

export default styles;
