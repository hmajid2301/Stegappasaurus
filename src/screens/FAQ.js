import PropTypes from 'prop-types';
import React, { Component } from 'react';
import {
  Dimensions,
  Text,
  TouchableOpacity,
  ScrollView,
  StyleSheet,
  View,
} from 'react-native';
import Accordion from 'react-native-collapsible/Accordion';
import { Icon } from 'react-native-elements';

import Header from '../components/Header';
import { colors, fonts } from '../util/styles';


const faq = [
  {
    title: 'What is steganography?',
    content: 'Steganography is the art/science of hiding data in plain sight. In the context of this app, it allows ' +
      'you to hide messages (text) within images. You do this by encoding the image with the text data. You can then share the image ' +
      'with other people and they can decode the image (retrieve the original message) using this app.',
  },
  {
    title: 'So is it the same as encryption?',
    content: 'Not quite, encryption involves obscuring information whereas steganography is concerned with hiding it. ' +
      'With encryption if you have the encrypted message you cannot retrieve the original message. \n\n' +

      'Whereas with steganography if you know where to look you can retrieve the original message, it is not obscured in any way. ' +
      'However, with encryption it is obvious you are trying to hide something but with steganography you can hide your message inside ' +
      'of an "innocent" looking image, with a third party without anyone noticing.\n\n' +

      'You can combine the two "techniques", encrypt your data before encoding it into an image, then simply decrypt the' +
      'message after decoding it from the image, ' +
      'for added security. This involves a shared secret "password" between you and the person you are sending the image to.',
  },
  {
    title: 'How does this app work?',
    content: 'This app uses different steganography algorithms to encode the RGB (Red, Green, Blue) pixel values with the text ' +
      'data in such a way that the original data can be retrieved at a later date.\n\n' +

      'After the image has been encoded you can save the image, then you can share this image and you use this app ' +
      'to decode and get the originally encoded message.\n\n' +

      'In general each algorithm will encode the size of the message, this is so that when decoding the image we know when to stop. ' +
      'It will then encode a separator (Â£ in ASCII), this is so that again when we decode we know from this point to decode n characters. ' +
      'Each character in the message usually uses 8 bits in the images, this is because we encode it as a the ASCII representation of the letter or symbol.',
  },
  {
    title: 'What Algorithms can you use?',
    content: 'You can use the following algorithms to encode/decode images: \n\n' +
              'Bitmap Least Significant Bit (LSB) \n' +
              'Discrete Cosine Transform (DCT) Least Significant Bit (LSB) \n',
  },
  {
    title: 'How much data can I encode with LSB?',
    content: 'It all depends on the image resolution, the bigger the image the more data you can encode. ' +
      'To find out if your message will fit in the image you should solve Equation 1 shown below where,' +
      '\n\n h = Image Height (in Pixels), \n w = Image Width (in pixels) and \n m is the number of characters in your message\n\n' +

      'To find the maximum number of characters we can encode we must use the same equation above except we = 0 not > 0 and we will solve for "m". ' +
      'Let\'s say our image is 100 x 100, then we have 100 * 100 * 3 = 30,000 - 8 (for separator) = 29,992. So 29,992 = ceil(log2(m)) - 8m = 3747. ' +
      'So using the LSB algorithm we can encode a message with 3747 characters.\n\n' +
      'Note: With LSB we only can encode one bit per byte of data, so that we don\'t completely change the colour of that pixel.' +
      'Equation 1: 3(h * w) - ceil(log2(m)) - 8m -8 > 0',
  },
  {
    title: 'Can you show me an example?',
    content: 'We will use an image 100 x 100 = 10,000 pixels. So 10,000 * 3 * 8 = 30,000 bits we can encode.' +
      'Let\'s say we want to encode "Hello World!", it\'s of length 12 so ceil(log2(12)) = 4 bits for the length of the message. ' +
      'So  4 + 8 = 12 bits required so far. Each character we encode will take 8 bits and we have 12 characters ' +
      '(including blank space) 12 * 8 = 96 bits for the message, so in total 96 + 12 = 108 bits ' +
      'would be required out of 30,000, to encode the message "Hello World!".',
  },
  {
    title: 'Why can we only use one bit out of every byte?',
    content: 'If we used the entire byte, remember per pixel we get 3 bytes (one for each RGB item), we would drastically ' +
      'change the colour of image.',
  },
];

const styles = StyleSheet.create({
  sectionHeaderContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    height: 50,
    borderBottomColor: colors.pureWhite,
    borderBottomWidth: 1,
  },
  active: {
    backgroundColor: colors.primary,
  },
  inactive: {
    backgroundColor: colors.secondary,
  },
  sectionHeader: {
    color: colors.pureWhite,
    fontFamily: fonts.body,
  },
  sectionContentContainer: {
    paddingHorizontal: 20,
    paddingVertical: 20,
  },
  sectionContent: {
    fontFamily: fonts.body,
  },
  container: {
    flex: 1,
    flexDirection: 'column',
  },
  accordionContainer: {
    alignItems: 'stretch',
    paddingTop: 20,
  },
  image: {
    width: (Dimensions.get('window').width - 40),
    height: 200,
    resizeMode: 'contain',
  },
});


export default class FAQ extends Component {
  static navigationOptions = {
    drawerLabel: 'FAQ',
    drawerIcon: ({ tintColor }) => (
      <Icon name='question-circle' type='font-awesome' color={tintColor}/>
    ),
  };

  static propTypes = {
    navigation: PropTypes.object.isRequired,
  };

  state = {
    activeSections: [],
    collapsed: true,
  };

  toggleExpanded = () => {
    this.setState({ collapsed: !this.state.collapsed });
  };

  setSections = (sections) => {
    this.setState({
      activeSections: sections.includes(undefined) ? [] : sections,
    });
  };

  renderHeader = (section, _, isActive) => (
    <View style={[styles.sectionHeaderContainer, isActive ? styles.inactive : styles.active]}>
      <Text style={styles.sectionHeader}>{section.title}</Text>
    </View>
  );

  renderContent = section => (
    <View style={styles.sectionContentContainer}>
      <Text style={styles.sectionContent}>{section.content}</Text>
    </View>
  );

  render() {
    return (
      <View style={styles.container}>
        <ScrollView>
          <Header color={colors.primary} navigation={this.props.navigation}/>

          <View style={styles.accordionContainer}>
            <Accordion
              activeSections={this.state.activeSections}
              duration={200}
              onChange={this.setSections}
              renderContent={this.renderContent}
              renderHeader={this.renderHeader}
              sections={faq}
              touchableComponent={TouchableOpacity}
            />
          </View>
        </ScrollView>
      </View>
    );
  }
}
