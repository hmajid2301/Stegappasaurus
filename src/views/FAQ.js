import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { ScrollView, View } from 'react-native';
import { Content, Icon } from 'native-base';

import CustomHeader from '~/components/CustomHeader';
import AccordionList from '~/views/FAQ/AccordionList';
import { colors } from '~/util/styles';

import faq from './FAQ/questions';
import styles from './FAQ/styles';


export default class FAQ extends Component {
  static navigationOptions = {
    drawerLabel: 'FAQ',
    drawerIcon: ({ tintColor }) => (
      <Icon name='questioncircleo' type='AntDesign' style={{ color: tintColor }}/>
    ),
  };

  static propTypes = {
    navigation: PropTypes.object.isRequired,
    screenProps: PropTypes.object.isRequired,
  };

  render() {
    const { theme } = this.props.screenProps;

    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <ScrollView>
          <CustomHeader
            color={colors.primary}
            navigation={this.props.navigation}
            theme={theme}
          />

          <View style={styles.accordionContainer}>
            <Content padder>
              <AccordionList faq={faq} theme={theme}/>
            </Content>
          </View>
        </ScrollView>
      </View>
    );
  }
}
