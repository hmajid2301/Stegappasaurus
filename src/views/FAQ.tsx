import * as React from "react";
import { ScrollView, View } from "react-native";
import { NavigationScreenProp } from "react-navigation";

import { ITheme, PrimaryColor } from "@types";
import AppHeader from "~/components/AppHeader";
import FAQList from "~/components/FAQList";
import { colors } from "~/constants";

import faq from "./FAQ/questions";
import styles from "./FAQ/styles";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
  screenProps: {
    theme: ITheme;
  };
}

export default class FAQ extends React.Component<IProps, {}> {
  public render() {
    const { theme } = this.props.screenProps;

    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <ScrollView>
          <AppHeader
            primaryColor={colors.primary as PrimaryColor}
            navigation={this.props.navigation}
            theme={theme}
          />

          <View style={styles.faqListContainer}>
            <FAQList items={faq} theme={theme} />
          </View>
        </ScrollView>
      </View>
    );
  }
}
