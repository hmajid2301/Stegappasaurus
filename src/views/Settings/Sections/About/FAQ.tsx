import * as React from 'react';
import {ScrollView, View} from 'react-native';

import FAQList from '~/components/FAQList';
import {ThemeColors} from '~/constants/types';
import {questions} from '~/data';
import styles from './FAQ/styles';

interface IProps {
  background: ThemeColors;
  color: ThemeColors;
}

const FAQ = (props: IProps) => (
  <View style={[styles.container, {backgroundColor: props.background}]}>
    <ScrollView>
      <View style={styles.faqListContainer}>
        <FAQList
          backgroundColor={props.background}
          color={props.color}
          items={questions}
        />
      </View>
    </ScrollView>
  </View>
);

export default FAQ;
