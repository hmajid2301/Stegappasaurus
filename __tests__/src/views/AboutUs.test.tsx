import { shallow } from "enzyme";
import React from "react";

import FAQ from "~/views/FAQ";
describe("FAQ: Match Snapshots", () => {
  test("1", () => {
    const component = shallow(
      <FAQ
        navigation={jest.fn() as any}
        screenProps={{
          theme: {
            color: "#17212D",
            background: "#FFF",
            isDark: false
          }
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test("2", () => {
    const component = shallow(
      <FAQ
        navigation={jest.fn() as any}
        screenProps={{
          theme: {
            background: "#17212D",
            color: "#FFF",
            isDark: true
          }
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
