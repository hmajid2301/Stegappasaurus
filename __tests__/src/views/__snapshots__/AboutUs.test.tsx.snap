// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ: Match Snapshots 1 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
      },
      Object {
        "backgroundColor": "#FFF",
      },
    ]
  }
>
  <ScrollViewMock>
    <AppHeader
      navigation={[MockFunction]}
      primaryColor="#009CFF"
      theme={
        Object {
          "background": "#FFF",
          "color": "#17212D",
          "isDark": false,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "stretch",
          "paddingTop": 5,
        }
      }
    >
      <FAQList
        backgroundColor="#FFF"
        color="#17212D"
        items={
          Array [
            Object {
              "content": "Steganography is the art/science of hiding data in plain sight. In the context of this app, it allows you to hide messages (text) within images. You do this by encoding the image with the text data. You can then share the image with other people and they can decode the image (retrieve the original message) using this app.",
              "title": "What is steganography?",
            },
            Object {
              "content": "Not quite, encryption, involves obscuring information whereas steganography is concerned with hiding it. With encryption if you have the encrypted message you cannot retrieve the original message. 

Whereas with steganography if you know where to look you can retrieve the original message, it is not obscured in any way. However, with encryption, it is obvious you are trying to hide something but with steganography you can hide your message inside of an \\"innocent\\" looking image, with a third party without anyone noticing.

You can combine the two \\"techniques\\", encrypt your data before encoding it into an image, then simply decrypt themessage after decoding it from the image, for added security. This involves a shared secret \\"password\\" between you and the person you are sending the image to.",
              "title": "So is it the same as encryption?",
            },
            Object {
              "content": "This app uses different steganography algorithms to encode the RGB (Red, Green, Blue) pixel values with the text data in such a way that the original data can be retrieved at a later date.

After the image has been encoded you can save the image, then you can share this image and you use this app to decode and get the originally encoded message.

In general each algorithm will encode the size of the message, this is so that when decoding the image we know when to stop. Each character in the message usually uses 8 bits in the images, this is because we encode it using the UTF-8 representation of the letter, symbol or emoji.",
              "title": "How does this app work?",
            },
            Object {
              "content": "There are many ways to represent colours in an image one of the most common ways is to the (8 bit) RGB colour system, which is red, green and blue. Each pixel has three components a red component, a green component and a blue component. Each component can be set a value from 0 to 255 (one byte). In binary 0 as a bytewould be represented by 00000000 and a value of 255 would be represented by 1111111. It is often referred to as 8 bit RGB because each component can only be represented by eight 1s or 0s (also referred to as a byte). So for example, if the RGB values of a pixel was set to 255 (all of them) then that pixel would be white (all colours added together). However, if the RGB values were set to 0, the pixel would be black. By changing the values of the RGB components we get different colours. So, all in all, we can represent 255 ^ 3 colours in 8-bit RGB or 16,777,216 different colours.",
              "title": "How do RGB images work?",
            },
            Object {
              "content": "LSB which stands for least significant bits, involves changing the smallest bit in a binary number. The best way to understand LSB is to look at an examplelet's say we have the letter A we want to encode into a 100 by 100 image which is all white. Each pixel in that image will have RGBA values of 255, remember thatfor each pixel in the image, we can encode 3 bits into since each pixel has three values which determine it's colour Red, Green, Blue. Technically we also havean Alpha value but ignore it in our case.

So in our example we 100 * 100 * 3 = 30000, bits we can encode. Now before we can encode 'A' we need to convert it to a number first. Luckily we can use UTF-8where A becomes the number 65. So we will encode 65 and then when we decode it from the image we will know (using UTF-8) that this 65 is an A. Before wecan encode 65, we need to convert it into a byte which is the binary representation of 65. Which is 01000001, so now we go through this number one at a time andencode each bit into our image. So first we have '0' and we want to encode that into the Red component of our first pixel is 255 which in binary is 11111111. Now We want to change the LSB which is the rightmost bit since '1' is not equal to '0', we will encode a '0' so our new Red component becomes 11111110 which is 254.Notice how the number only decreased by 1. So there won't be a large colour difference in the image between 255 and 254 (Red). However, if we changed the mostsignificant bit (leftmost bit) it would become 01111111 which is 127, which would be a much large colour difference. Which is why we always change the LSB. So to include65 we will need 8 bytes so we will encode the following components RGB RGB RG (across 3 pixels).In our specific implementation, we also include the size of the message we are encoding at the beginning so in this case 'A' message is just 1. So we will encode the following byte '00000001'. So the whole message that we will encode is '0000000101000001'.",
              "title": "How does LSB work?",
            },
          ]
        }
      />
    </View>
  </ScrollViewMock>
</View>
`;

exports[`FAQ: Match Snapshots 2 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
      },
      Object {
        "backgroundColor": "#17212D",
      },
    ]
  }
>
  <ScrollViewMock>
    <AppHeader
      navigation={[MockFunction]}
      primaryColor="#009CFF"
      theme={
        Object {
          "background": "#17212D",
          "color": "#FFF",
          "isDark": true,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "stretch",
          "paddingTop": 5,
        }
      }
    >
      <FAQList
        backgroundColor="#17212D"
        color="#FFF"
        items={
          Array [
            Object {
              "content": "Steganography is the art/science of hiding data in plain sight. In the context of this app, it allows you to hide messages (text) within images. You do this by encoding the image with the text data. You can then share the image with other people and they can decode the image (retrieve the original message) using this app.",
              "title": "What is steganography?",
            },
            Object {
              "content": "Not quite, encryption, involves obscuring information whereas steganography is concerned with hiding it. With encryption if you have the encrypted message you cannot retrieve the original message. 

Whereas with steganography if you know where to look you can retrieve the original message, it is not obscured in any way. However, with encryption, it is obvious you are trying to hide something but with steganography you can hide your message inside of an \\"innocent\\" looking image, with a third party without anyone noticing.

You can combine the two \\"techniques\\", encrypt your data before encoding it into an image, then simply decrypt themessage after decoding it from the image, for added security. This involves a shared secret \\"password\\" between you and the person you are sending the image to.",
              "title": "So is it the same as encryption?",
            },
            Object {
              "content": "This app uses different steganography algorithms to encode the RGB (Red, Green, Blue) pixel values with the text data in such a way that the original data can be retrieved at a later date.

After the image has been encoded you can save the image, then you can share this image and you use this app to decode and get the originally encoded message.

In general each algorithm will encode the size of the message, this is so that when decoding the image we know when to stop. Each character in the message usually uses 8 bits in the images, this is because we encode it using the UTF-8 representation of the letter, symbol or emoji.",
              "title": "How does this app work?",
            },
            Object {
              "content": "There are many ways to represent colours in an image one of the most common ways is to the (8 bit) RGB colour system, which is red, green and blue. Each pixel has three components a red component, a green component and a blue component. Each component can be set a value from 0 to 255 (one byte). In binary 0 as a bytewould be represented by 00000000 and a value of 255 would be represented by 1111111. It is often referred to as 8 bit RGB because each component can only be represented by eight 1s or 0s (also referred to as a byte). So for example, if the RGB values of a pixel was set to 255 (all of them) then that pixel would be white (all colours added together). However, if the RGB values were set to 0, the pixel would be black. By changing the values of the RGB components we get different colours. So, all in all, we can represent 255 ^ 3 colours in 8-bit RGB or 16,777,216 different colours.",
              "title": "How do RGB images work?",
            },
            Object {
              "content": "LSB which stands for least significant bits, involves changing the smallest bit in a binary number. The best way to understand LSB is to look at an examplelet's say we have the letter A we want to encode into a 100 by 100 image which is all white. Each pixel in that image will have RGBA values of 255, remember thatfor each pixel in the image, we can encode 3 bits into since each pixel has three values which determine it's colour Red, Green, Blue. Technically we also havean Alpha value but ignore it in our case.

So in our example we 100 * 100 * 3 = 30000, bits we can encode. Now before we can encode 'A' we need to convert it to a number first. Luckily we can use UTF-8where A becomes the number 65. So we will encode 65 and then when we decode it from the image we will know (using UTF-8) that this 65 is an A. Before wecan encode 65, we need to convert it into a byte which is the binary representation of 65. Which is 01000001, so now we go through this number one at a time andencode each bit into our image. So first we have '0' and we want to encode that into the Red component of our first pixel is 255 which in binary is 11111111. Now We want to change the LSB which is the rightmost bit since '1' is not equal to '0', we will encode a '0' so our new Red component becomes 11111110 which is 254.Notice how the number only decreased by 1. So there won't be a large colour difference in the image between 255 and 254 (Red). However, if we changed the mostsignificant bit (leftmost bit) it would become 01111111 which is 127, which would be a much large colour difference. Which is why we always change the LSB. So to include65 we will need 8 bytes so we will encode the following components RGB RGB RG (across 3 pixels).In our specific implementation, we also include the size of the message we are encoding at the beginning so in this case 'A' message is just 1. So we will encode the following byte '00000001'. So the whole message that we will encode is '0000000101000001'.",
              "title": "How does LSB work?",
            },
          ]
        }
      />
    </View>
  </ScrollViewMock>
</View>
`;
